options_test.go

```diff
@@ -70,3 +107,61 @@ func TestCannotSetFiltersAndConnGater(t *testing.T) {
 	require.Error(err)
 	require.Contains(err.Error(), "cannot configure multiple connection gaters")
 }
+
+/// NOTE(jalextowle): This was taken directly from https://github.com/0x-mesh/p2p/banner/banner.go
```

Comment:
@jalextowle 

You can use this instead:

https://github.com/multiformats/go-multiaddr/blob/d18c05e0e1635f8941c93f266beecaedd4245b9f/net/convert.go#L103